### PR TITLE
[Enhancement] Support is_role_in_session function

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 20 / 22 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|16|18|88.89%|[1346, 1347]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java [hunk 1]
```diff
@@ -1326,5 +1331,34 @@ public static ConstantOperator urlExtractParameter(ConstantOperator url, Constan
...
+            }
+        } catch (PrivilegeException e) {
+            throw new SemanticException(e.getMessage());
+        }
+    }
+
+    @ConstantFunction(name = "is_role_in_session", argTypes = {VARCHAR}, returnType = BOOLEAN)
+    public static ConstantOperator isRoleInSession(ConstantOperator role) {
+        AuthorizationMgr manager = GlobalStateMgr.getCurrentState().getAuthorizationMgr();
+        List<String> roleName = new ArrayList<>();
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java:1354]: roleName -> roleNames, besides using hash set is better i think.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java [hunk 2]
```diff
@@ -1326,5 +1331,34 @@ public static ConstantOperator urlExtractParameter(ConstantOperator url, Constan
         }
         return ConstantOperator.createNull(Type.VARCHAR);
     }
+
+    private static void getRecursiveRole(AuthorizationMgr manager, List<String> roleName, Long roleId) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java:1335]: This is pretty much like `protected void getAllPredecessorsInner(long roleId, Set<Long> resultSet) throws PrivilegeException `,  how about renaming it to `getAllPredecessorNames()` and put it in `AuthorizationMgr`, it maybe useful in other situations.

**Review by nshangyiming — COMMENTED** [review]
comments

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**